kitsune/wiki/utils.py

```diff
@@ -116,10 +124,35 @@ def _active_contributors_id(from_date, to_date, locale, product):
 
     if product:
         editors = editors.filter(
-            Q(document__products=product) |
-            Q(document__parent__products=product))
+            Q(document__products=product) | Q(document__parent__products=product)
+        )
         reviewers = reviewers.filter(
-            Q(document__products=product) |
-            Q(document__parent__products=product))
+            Q(document__products=product) | Q(document__parent__products=product)
+        )
 
     return set(list(editors) + list(reviewers))
+
+
+def get_featured_articles(product=None):
+    """Returns 4 random articles from the most visited.
+
+    If a product is passed, it returns 4 random highly visited articles.
+    """
+
+    featured = (
+        WikiDocumentVisits.objects.filter(period=LAST_7_DAYS)
+        .order_by("-visits")
+        .select_related("document")
+    )
+
+    if product:
+        featured = featured.filter(document__products__in=[product.id])
+
+    # Limit this to 10 entries
+    if featured.count() < 10:
+        return featured
+
+    # This returns a WikiDocumentVisits object and not the actual Wiki doc.
+    # The wiki doc it's under .document which is prefetched b/c of select_related
+    featured = featured[:10]
+    return random.sample(featured, 4)
```

Comment:
nit: Here you can use `order_by('?')` to get the random order

Reply:
You cannot reorder the queryset after you slicing thus the workaround with random.